* Задача 1. Даны две независимые выборки. Не соблюдается условие нормальности
x1 380,420, 290
y1 140,360,200,900
Сделайте вывод по результатам, полученным с помощью функции, имеются ли статистические различия между группами?

In [2]:
from scipy import stats
import numpy as np

In [3]:
x_1 = np.array([380,420, 290 ])
y_1 = np.array([140,360,200,900])
alpha = 0.05

In [4]:
stats.mannwhitneyu(x_1, y_1)

MannwhitneyuResult(statistic=8.0, pvalue=0.6285714285714286)

pvalue > alpha \
статистические различия между группами __нет__

* Задача 2. Исследовалось влияние препарата на уровень давления пациентов. Сначала
измерялось давление до приема препарата, потом через 10 минут и через 30 минут. Есть
ли статистически значимые различия между измерениями давления? В выборках не соблюдается условие нормальности.
1е измерение до приема препарата: 150, 160, 165, 145, 155
2е измерение через 10 минут: 140, 155, 150, 130, 135
3е измерение через 30 минут: 130, 130, 120, 130, 125

In [5]:
g_1 = np.array([150, 160, 165, 145, 155])
g_2 = np.array([140, 155, 150, 130, 135])
g_3 = np.array([130, 130, 120, 130, 125])
alpha = 0.05

In [6]:
stats.friedmanchisquare(g_1, g_2, g_3)

FriedmanchisquareResult(statistic=9.578947368421062, pvalue=0.00831683351100441)

pvalue < alpha \
статистически значимые различия между измерениями давления __есть__

* Задача 3. Сравните 1 и 2 е измерения, предполагая, что 3го измерения через 30 минут не было. Есть
ли статистически значимые различия между измерениями давления?

In [7]:
alpha = 0.05
stats.wilcoxon(g_1, g_2)

WilcoxonResult(statistic=0.0, pvalue=0.0625)

pvalue > alpha \
статистически значимые различия между измерениями давления __нет__

* Задача 4. Даны 3 группы учеников плавания. Не соблюдается условие нормальности.
В 1 группе время на дистанцию 50 м составляют:
56, 60, 62, 55, 71, 67, 59, 58, 64, 67
Вторая группа : 57, 58, 69, 48, 72, 70, 68, 71, 50, 53
Третья группа: 57, 67, 49, 48, 47, 55, 66, 51, 54
Есть
ли статистически значимые различия между группами?

In [9]:
g_1 = np.array([56, 60, 62, 55, 71, 67, 59, 58, 64, 67])
g_2 = np.array([57, 58, 69, 48, 72, 70, 68, 71, 50, 53])
g_3 = np.array([57, 67, 49, 48, 47, 55, 66, 51, 54])
alpha = 0.05

In [10]:
stats.kruskal(g_1, g_2, g_3)

KruskalResult(statistic=5.465564058257224, pvalue=0.0650380998590494)

pvalue > alpha \
статистически значимые различия между группами __нет__

* Задача 5. Заявляется, что партия изготавливается со средним арифметическим 2,5 см. Проверить
данную гипотезу, если известно, что размеры изделий подчинены нормальному закону
распределения. Объем выборки 10, уровень статистической значимости 5%
2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34

In [18]:
alpha = 0.05
n = 10
stats.norm.ppf(alpha / 2) # выбрал z критерий так как посчитал, что "партия " -это генеральная совокупность

-1.9599639845400545

In [20]:
stats.norm.ppf(1 - alpha / 2) # выбрал z критерий так как посчитал, что "партия " -это генеральная совокупность

1.959963984540054

In [21]:
set_ditail = np.array([2.51, 2.35, 2.74, 2.56, 2.40, 2.36, 2.65, 2.7, 2.67, 2.34])

In [22]:
x_mean = np.mean(set_ditail)
x_mean

2.5279999999999996

In [23]:
x_var = np.var(set_ditail, ddof=1)
x_var

0.02472888888888891

In [24]:
z_h = (x_mean - x_var) / (2.5 / np.sqrt(n))
z_h

3.166415324804617

1.959963984540054 < 3.166415324804617 \
критическое значение меньше чем z_h (Партия __не__ изготавливается со средним арифметическим 2,5 см.)